sys/ztimer64/Makefile.dep

```diff
@@ -0,0 +1,37 @@
+#
+# ztimer64 dependencies
+#
+
+ifneq (,$(filter ztimer64_usec,$(USEMODULE)))
+  USEMODULE += ztimer64 ztimer_usec
```

Comment:
Here and below, this Makefile is included depending on `ztimer64`, so this is not needed.

```suggestion
  USEMODULE += ztimer_usec
```

Reply:
that's not correct, it is included depending on `ztimer64%`. So if `ztimer64_usec` causes the file to be included, it still needs to select `ztimer64`.

Reply:
I re-worked the dependencies a bit [here](https://github.com/fjmolinas/RIOT/tree/wip/ztimer64_deps), mainly I didn't like the auto_init stuff and I added Kconfig. let me know what you think..,

Reply:
+1

I've simplified the ztimer64 deps a bit more, otherwise pulled your branch. Thanks a bunch for the KConfig stuff!